src/adam_core/propagator/propagator.py

```diff
@@ -89,17 +93,158 @@ class EphemerisMixin:
     Subclasses should implement the _generate_ephemeris method.
     """
 
-    @abstractmethod
+    def _add_light_time(
+        self,
+        orbits,
+        observers,
+        lt_tol: float = 1e-12,
+        max_iter: int = 10,
```

Comment:
I had set this to 10 just because its a nice number but feel free to change this. Our examples from yesterday seemed to complete in about 4 iterations.

Reply:
I don't really have a sense of the long tail curve for convergence (if there even is one). In my mind, it's perfectly okay if the max_iter is literally never reached, it's only there to keep something that won't converge from running on forever.